src/GraphQL.Tests/Builders/ConnectionBuilderTests.cs

```diff
@@ -16,21 +16,11 @@ public void should_have_name()
     [Theory]
     [InlineData("")]
     [InlineData(null)]
-    public void should_throw_error_if_name_is_null_or_empty(string fieldName)
+    public void should_throw_error_if_name_is_null_or_empty(string? fieldName)
     {
         var type = new ObjectGraphType();
         ArgumentOutOfRangeException exception;
-        // race condition with does_not_throw_with_filtering_nameconverter test
-        try
-        {
-            exception = Should.Throw<ArgumentOutOfRangeException>(() => type.Connection<ObjectGraphType>(fieldName));
-        }
-        catch (ShouldAssertException)
-        {
-            System.Threading.Thread.Sleep(100); // wait a bit and retry
-            exception = Should.Throw<ArgumentOutOfRangeException>(() => type.Connection<ObjectGraphType>(fieldName));
-        }
```

Comment:
`does_not_throw_with_filtering_nameconverter` currently has `[Collection("StaticTests")]` defined on the class.  It should not have a race condition in the future with any code.  Very likely this was a patch that did not work and was subsequently fixed by adding the class to that collection.